crowbar_framework/app/models/api/storage.rb

```diff
@@ -15,6 +15,32 @@
 #
 
 module Api
-  class Storages
+  class Storage < Tableless
+    def repocheck
+      {}.tap do |ret|
+        ret["ceph"] = {}
+        ceph_node_architectures = NodeObject.find(
+          "roles:ceph-mon AND ceph_config_environment:*"
+        ).map(&:architecture).uniq
+
+        platform = Api::Upgrade.new.target_platform(platform_exception: :ceph)
+
+        ["os", "ceph"].each do |feature|
+          ceph_node_architectures.each do |architecture|
+            unless ::Crowbar::Repository.provided_and_enabled?(feature,
+                                                               platform,
+                                                               architecture)
+              ::Openstack::Upgrade.enable_repos_for_feature(feature, Rails.logger)
+            end
+            available, repolist = ::Crowbar::Repository.provided_and_enabled_with_repolist(
+              feature, platform, architecture
+            )
+            ret["ceph"]["available"] = available
+            ret["ceph"]["repos"] ||= {}
+            ret["ceph"]["repos"].deep_merge!(repolist.deep_stringify_keys)
```

Comment:
I guess you need to put the architecture somewhere in the hash, to report that to the user.